### PR TITLE
hexdecoct: make return parameters of unbase64mem() and unhexmem() opt…

### PR DESCRIPTION
**Pull request by poettering**
…ional

Inspired by: #19059

### CODE REVIEW

File: src/basic/hexdecoct.c [hunk 1]
```diff
@@ -115,8 +115,6 @@ int unhexmem_full(const char *p, size_t l, bool secure, void **ret, size_t *ret_
         uint8_t *z;
         int r;
 
-        assert(ret);
-        assert(ret_len);
```
Comment (bluca) [inline src/basic/hexdecoct.c:119]: Can these be used individually? Or would it make sense to assert that they are either both set or both null?

Comment (poettering) [reply src/basic/hexdecoct.c:119]: it definitely makes sense to call with ret == NULL and ret_len != NULL, if you just want to know the size of a base64 blob.

and with ret != NULL but ret == NULL also might make sense, if you already know the size ahead of time, and don't have to check again. example: you want to write the base64 data to a file, so you first call unhexmem() with ret = NULL to get the size you need. Now you create the file and fallocate() the right size for it, and then mmap () it. Then you call unhexmem() again, pointing directly to the mmap(), but this time you don't need the size anymore, you know it already...

(not saying the scenario is something i'd actually write that way, but i still think it's a valid scenario)

**Comment by yuwata** [comment]
LGTM.

File: src/basic/hexdecoct.c (hunk 1 above)
Comment (bluca) [reply src/basic/hexdecoct.c:119]: makes sense, thanks

**Review by bluca — APPROVED** [review]
lgtm

**Pull request merged**